### PR TITLE
overhaul software-info

### PR DESCRIPTION
**Pull request by owaddell**
This pull request obsoletes some entries in software-info that used to be baked into the executable in swish-win and generalizes the `software-revision` mechanism to track more information about other components.

@burgerrg, I'm still mulling over the last few commits, but would appreciate your input.

There are a few other things we might consider for the current round of testing improvements:
* `swish-test` should be writing the expected `.mo` files with some sort of "incomplete" mark so we report when something went completely sideways
* `swish-test` should have an option for generating a report from existing `.mo` files without running the tests.
* Chris suggested we consider flipping the `.mo` files over to JSON.

### CODE REVIEW

**Comment by burgerrg** [comment]
I didn't know about the `profile` macro. Nice work using it!

File: doc/swish/app.tex [hunk 1]
```diff
@@ -345,6 +345,89 @@ \subsection{Configuration}
...
+  \code{(software-revision \opt{\var{key}} \opt{\var{value}})}
+\end{parameter}
+\hasvalue{} a string or \code{\#f}
+
+The \code{software-revision} procedure stores or retrieves the software
+revision under the path \code{(\var{key} "revision")} in the
+\code{(software-info)} object.
+If \var{value} is omitted, this procedure returns the value, if any,
+associated with \var{key}, or else \code{\#f}.
+If \var{key} and \var{value} are omitted, this procedure returns the value,
```
Comment (burgerrg) [inline doc/swish/app.tex]: see comment above about the trailing comma

File: doc/swish/app.tex [hunk 2]
```diff
@@ -345,6 +345,89 @@ \subsection{Configuration}
...
+  \code{(software-version \opt{\var{key}} \opt{\var{value}})}
+\end{parameter}
+\hasvalue{} a string or \code{\#f}
+
+The \code{software-version} procedure stores or retrieves the software
+version under the path \code{(\var{key} "version")} in the
+\code{(software-info)} object.
+If \var{value} is omitted, this procedure returns the value, if any,
+associated with \var{key}, or else \code{\#f}.
+If \var{key} and \var{value} are omitted, this procedure returns the value,
```
Comment (burgerrg) [inline doc/swish/app.tex]: see comment above about the trailing comma

**Comment by owaddell** [comment]
@burgerrg I've made a few changes since 39df5c1, could you take another look?

- flipped cli-specs keys from strings to symbols
- documented inherited parameters
- folded Scheme Testing Library into swish.pdf
- added some tests
- added more mat meta-data
- moved the profile.ss changes to another branch
- changed the HTML mat-report a bit more

File: doc/swish/testing.tex [hunk 3]
```diff
@@ -114,10 +120,11 @@
 The \code{run-mats-to-file} procedure executes all mats and writes
 the results into the file specified by the string \var{filename}. If
 the files exists, its contents are overwritten. The file format is a
```
[the remark this answers is not in the document]
Comment (owaddell) [reply doc/swish/testing.tex]: Oops, missed this again in the last round of doc fixes. Good thing we have the resolve conversation workflow.

**Comment by owaddell** [comment]
@burgerrg thanks for all the reviewing! I aim to merge it this afternoon.

**Pull request merged**